src/views/docs-view/server.ts

```diff
@@ -260,27 +262,38 @@ export function getStaticGenerationFunctions<
 			}
 
 			/**
-			 * Constructs the levels of nav data used in the `Sidebar` on all
-			 * `DocsView` pages.
+			 * Constructs the base sidebar level for `DocsView`.
+			 */
+			const docsSidebarLevel: SidebarProps = {
+				backToLinkProps: {
+					text: `${product.name} Home`,
+					href: `/${product.slug}`,
+				},
+				levelButtonProps: {
+					levelUpButtonText: `${product.name} Home`,
+				},
+				menuItems: navDataWithFullPaths as EnrichedNavItem[],
+				// If the title is not hidden for this rootDocsPath, include it
+				title: BASE_PATHS_TO_NAMES[basePath] || product.name,
+			}
+			// If the title is not hidden for this rootDocsPath, include it
+			if (currentRootDocsPath.hideSidebarTitle) {
+				docsSidebarLevel.visuallyHideTitle = true
+			}
+			// If we want to add an "overview" item, add one
+			if (currentRootDocsPath.addOverviewItem) {
+				docsSidebarLevel.overviewItemHref = versionPathPart
+					? `/${product.slug}/${basePath}/${versionPathPart}`
+					: `/${product.slug}/${basePath}`
+			}
```

Comment:
AFAIK, we want the `overviewItemHref` behavior to be the default from a design standpoint, and removing it to be an exception. Other products depend on this behavior working as well, so this needs to be backwards compatible.

Could we flip this logic with a rename or with an adjustment to the conditional check? Maybe checking that the option has been explicitly turned off will work:

```typescript
// Add "Overview" item, unless explicitly disabled
if (currentRootDocsPath.addOverviewItem !== false) {
  docsSidebarLevel.overviewItemHref = versionPathPart
	  ? `/${product.slug}/${basePath}/${versionPathPart}`
		: `/${product.slug}/${basePath}`
}
```

Reply:
@ashleemboyer Definitely agreed that it seems like we want the `overviewItemHref` to be consistent across products. I think to get to that consistency, without code-level shims, I see two broad options:

1. _Remove_ all existing `"Overview"` items, for example from Terraform docs categories' `*-nav-data.json`, and rely on the implementation to shim in the desired content
2. _Add_ `"Overview"` items to all `*-nav-data.json` items where they're missing, and then remove our code that shims in this content

Personally I lean towards the `2.`, so I think we should have the implementation-side shimming of missing `"Overview"` items in sidebar content be the exception rather than the rule. So we should be explicitly setting `addOverviewItem` in cases where we haven't had the chance to make the content update.

That being said, I don't feel too strongly here - it just feels odd to me for us to try to implement a permanent-default fix (rather than temporary fix) for what seems to boil down to a content inconsistency issue. Let me know if that seems reasonable, happy to adjust to do the flip-logic-rename you suggest (say, `doNotAddOverviewItem`) if that still feels preferable 🙇‍♂️

Reply:
I think we have different perspectives of what the Overview item is. As it's currently coded, I see it as a feature of the platform, not a shim. If we were injecting this item server-side as we fetched/formatted the sidebar data, I'd see it more like a shim. But regardless of what we call it, isn't it easier for us to code this into the platform and maintain it rather than rely on 9 different teams and repositories to do it?

Our current approach that we've been following for a while is that we place an Overview item in the sidebar. These changes break that approach for other products, so I definitely think they need to be made in a backwards-compatible way. If we want to revisit the Overview item approach, I think it's a larger discussion we need to involve both design and product in.

Reply:
Can we detect whether or not there is an existing `Overview` item and add it if missing? Considering the constraints we have here:
* We want an `Overview` nav item for every root path
* Some nav data files have `Overview` items already defined
* We don't want to duplicate the nav items on older versions

Doing a conditional "shim" feels like the more robust approach. That also gives us the ability to shift to making this an authoring concern, if that is a route we want to take in the future.

Reply:
@ashleemboyer I think it is easier for us to code the `"Overview"` item into the platform, _if_ we have full author alignment on _not_ including the item. To me getting author alignment either way is necessary, so it seems logical to avoid extra code and instead ask authors to align content by including the `"Overview"` item rather than omitting it.

@BRKalow I'd be on board with a conditional shim. It seems like that'd be a more automatic version of manually specifying `addOverviewItem` in product data. Would we be alright with handling that in a separate PR? (Happy to tackle here in this PR if we'd rather; personally I lean towards a separate ticket and coming back to it after Terraform work is done).